NitroxPatcher/Patches/Dynamic/BreakableResource_BreakIntoResources_Patch.cs

```diff
@@ -0,0 +1,33 @@
+using HarmonyLib;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public class BreakableResource_BreakIntoResources_Patch : NitroxPatch, IDynamicPatch
+{
+    private static MethodInfo TARGET_METHOD = Reflect.Method((BreakableResource t) => t.BreakIntoResources());
+
+    public static void Prefix(BreakableResource __instance)
+    {
+        // Gets the entity ID
+        NitroxId id = NitroxEntity.GetId(__instance.gameObject);
```

Comment:
Should be changed to an if clause with NitroxEntity.TryGetEntityFrom, and only send the packet if we found an entity on the object because there would be no point in creating a new id then asking the server to delete it as if it always existed.

Reply:
Since the breakable entity already should have a Nitrox ID it shouldn't assign a new one (`GetId` does assing one if there is no ID set). 

Here's an example:

```cs
        if (!NitroxEntity.TryGetEntityFrom(__instance.gameObject, out NitroxEntity destroyedEntity))
        {
            Log.Warn($"[{nameof(BreakableResource_BreakIntoResources_Patch)}] Could not find {nameof(NitroxEntity)} for breakable entity {__instance.gameObject.GetFullHierarchyPath()}.");
            return;
        }

        Resolve<IPacketSender>().Send(new EntityDestroyed(destroyedEntity.Id));
```

Reply:
I still don't get the point, like why do you do that ?

Reply:
Code shouldn't do something unexpected like pretending a broken state (missing id) is fine and just assigning an unsynced id (only your client knows it) to the breakable entity.

Rather, we want to know that the state is unexpected. Because it means the server did a bad job synchronizing the id of the breakable entity to the client. Hence the logging and doing nothing (since there's no good way forward when the id missing).

Reply:
Hypothetically: in game it would leave things unsynchronized because we can't fix it client side. Meaning the breakable entity isn't broken but stays. The issue would be on the server and would require further investigation by us.

Reply:
like this ?
```cs
    public static void Prefix(BreakableResource __instance)
    {
        // Don't ask me what it does...
        if (!NitroxEntity.TryGetEntityFrom(__instance.gameObject, out NitroxEntity destroyedEntity))
        {
            Log.Warn($"[{nameof(BreakableResource_BreakIntoResources_Patch)}] Could not find {nameof(NitroxEntity)} for breakable entity {__instance.gameObject.GetFullHierarchyPath()}.");
            return;
        }
        // Send packet to destroy the entity
        Resolve<IPacketSender>().Send(new EntityDestroyed(destroyedEntity.Id));
    }
```

Reply:
Yes, that's all